s3po/backends/s3.py

```diff
@@ -93,3 +93,9 @@ def func():
             multi.upload_part_from_file(StringIO(data), count)
             multi.complete_upload()
             return True
+
+    def list(self, bucket, prefix=None, delimiter=None, headers=None):
```

Comment:
It seems like this should probably take a limit since some buckets can be extremely long, and we don't necessarily want to materialize the full list if we don't have to.